include/dr/mp/algorithms/count.hpp

```diff
@@ -0,0 +1,159 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+namespace dr::mp::__detail {
+
+inline auto add_counts(rng::forward_range auto &&r) {
+  rng::range_difference_t<decltype(r)> zero{};
+
+  return std::accumulate(rng::begin(r), rng::end(r), zero);
+}
+
+inline auto std_count_if(rng::forward_range auto &&r, auto &&pred) {
+  using count_type = rng::range_difference_t<decltype(r)>;
+
+  if (rng::empty(r)) {
```

Comment:
This conditional is redundant in std and dpl version. Please do the check before deciding between std and dpl.
Then you have only single call within std/dpl_count_if - remove the functions, do it inline